src/psyclone/parse/module_manager.py

```diff
@@ -222,21 +243,145 @@ def ignores(self):
         '''
         return self._ignore_modules
 
-    # ------------------------------------------------------------------------
-    def get_module_info(self, module_name):
-        '''This function returns the ModuleInformation for the specified
+    def add_files(self, filepaths: Union[str, List[str], Set[str]]) -> None:
+        """Add a file to the list of files
+
+        :param filepaths: Filename or list of filenames
+        """
+
+        if isinstance(filepaths, str):
+            filepaths = [filepaths]
+
+        for filepath in filepaths:
+            if filepath in self._filepath_to_file_info:
+                # Already loaded => skip
+                continue
+
+            self._filepath_to_file_info[filepath] = FileInfo(
+                filepath,
+                cache_active=self._cache_active,
+                cache_path=self._cache_path,
+            )
+
+    def load_all_source_codes(self, verbose: bool = False) -> None:
+        """Routine to load the source of all files previously added
+        to the module manager
+
+        :param verbose: If `True`, print verbose information
+        """
+
+        for fileinfo in self._filepath_to_file_info.values():
+            fileinfo: FileInfo
+            fileinfo.get_source_code(verbose=verbose)
+
+    def load_all_fparser_trees(self, verbose: bool = False) -> None:
```

Comment:
Presumably, this parses source as required, rather than just loading from file? Could therefore be `create_all_fparser_trees`?